add doc for workload identity (#3770)
kubernetes-sigs/cluster-api-provider-azure: docs/book/src/topics/multitenancy.md

To enable single controller multi-tenancy, a different Identity can be added to the Azure Cluster that will be used as the Azure Identity when creating Azure resources related to that cluster.

> This had been achieved using the [aad-pod-identity](https://azure.github.io/aad-pod-identity) library and is dperecated now and [workload identity](https://azure.github.io/azure-workload-identity/docs/) is the next iteration of it.

Reviewer: ```suggestion
This is achieved using [workload identity](https://azure.github.io/azure-workload-identity). Workload identity is the next iteration of the now deprecated [aad-pod-identity](https://azure.github.io/aad-pod-identity).
```

> Note that `Service Principal With Client Password` and `Service Principal With Certificate` identitites are based on AAD pod identity which is decsribed in the below sections.

Reviewer: ```suggestion
```

Reviewer: I would remove this in favor of labelling each section title

## Identity Types

> ### Workload Identity

Reviewer: ```suggestion
### Workload Identity (Recommended)
```

> Follow this [link](./workload-identity.md) for workload identity docs.

Reviewer: ```suggestion
Follow this [link](./workload-identity.md) for a quick start guide on setting up workload identity.

Once you've set up the management cluster with the workload identity (see link above), the corresponding values should be used to create an `AzureClusterIdentity` resource. Create an `azure-cluster-identity.yaml` file with the following content:

```yaml
apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
kind: AzureClusterIdentity
metadata:
  name: cluster-identity
spec:
  type: WorkloadIdentity
  allowedNamespaces: {}
  tenantID: <your-tenant-id>
  clientID: <your-client-id>
  allowedNamespaces: 
    list:
    - <cluster-namespace>
```

Reviewer: I think re-iterating what a workload identity `AzureClusterIdentity` looks like in this doc will make sure users don't miss it and go directly to the other example yamls below

> ### Service Principal With Client Password

Reviewer: ```suggestion
### AAD Pod Identity using Service Principal With Client Password (Deprecated)
```

Reviewer: I can't add comments on the lines below but should be the same for other sections

`### AAD Pod Identity using Service Principal With Certificate (Deprecated)`
`### AAD Pod Identity using User-Assigned Managed Identity (Deprecated)`

Author: Sure, but for the following:

> ### AAD Pod Identity using User-Assigned Managed Identity (Deprecated)

I don't think User assigned managed identity as anything to do with AAD pod identity.

Reviewer: it does, it uses AAD pod identity too https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/scope/identity.go#L169 

https://azure.github.io/aad-pod-identity/docs/concepts/azureidentity/#azureidentityspec

Author: Thanks.

## Identity Types

### Workload Identity

> AAD pod identity is deprecated and workload identity is the next iteration of it.

Reviewer: this contradicts the line right above (line 5) which says `this is achieved using the [aad-pod-identity](https://azure.github.io/aad-pod-identity) library.`

I also think it's unclear which ones below are AAD pod identity and which ones aren't.

Author: Have tried to incorporate this comment. Let me know if it looks good or any more changes are required.

Reviewer: I added some more specific comments with what I had in mind, let me know if it makes sense.

I really think the resulting docs need to be unambiguously pointing to workload identity and making it very clear that AAD pod identity is deprecated to avoid confusion.

  clientSecret: <client-secret-of-SP-identity>
```

> ### AAD Pod Identity using Service Principal With Certificate

Reviewer: ```suggestion
### AAD Pod Identity using Service Principal With Certificate (Deprecated)
```

```

For more details on how aad-pod-identity works, please check the guide [here](https://azure.github.io/aad-pod-identity/docs/).

> ### Workload Identity

Author: Done

> Follow this [link](#workload-identity) for workload identity docs.

Reviewer 2: ```suggestion
Follow this [link](./workload-identity.md) for workload identity docs.
```
AFAICT you need to make this change *and* add it to the ToC in `SUMMARY.md` for the doc to be visible.

Author: Done. Thanks

```

> For more details on how aad-pod-identity works, please check the guide [here](https://azure.github.io/aad-pod-identity/docs/).

Reviewer: ```suggestion
For more details on how workload identity works, please check the guide [here](https://azure.github.io/azure-workload-identity/docs/).
```

Reviewer: or just remove this line if it's redundant information

Reviewer: Line 158 should also be changed from 

```
The rest of the configuration is the same as that of service principal identity. This useful in scenarios where you don't want to have a dependency on [aad-pod-identity](https://azure.github.io/aad-pod-identity).
```

to 

```
The rest of the configuration is the same as that of the service principal identity. This is useful in scenarios where you don't want to have a dependency on [workload identity](https://azure.github.io/azure-workload-identity).
```

Author: I think, Manual service principal should not have a dependency on either workload identity or aad pod identity and infact it does not have any dependency.

Author: So removing that line makes sense.
